### PR TITLE
extension of metrics test with respect to new version 2.0 style

### PR DESCRIPTION
**Pull request by penehyba**

### CODE REVIEW

**Review by arjantijms — APPROVED** [review]

File: metrics/base-metrics/src/test/java/org/eclipse/microprofile12/metrics/base/BaseMetricsTest.java [hunk 1]
```diff
@@ -55,10 +55,10 @@ public void testBasePrometheusMetrics() throws IOException {
                 .request(TEXT_PLAIN)
                 .get();
         String responseText = response.readEntity(String.class);
-        assertTrue(responseText.contains("# TYPE base:thread_max_count"));
-        assertTrue(responseText.contains("# TYPE base:thread_count"));
-        assertTrue(responseText.contains("# TYPE base:cpu_system_load_average"));
-        assertTrue(responseText.contains("# TYPE base:jvm_uptime"));
+        assertTrue(responseText.contains("# TYPE base:thread_max_count") || responseText.contains("# TYPE base_thread_max_count gauge"));
```
Comment (Ladicek) [inline metrics/base-metrics/src/test/java/org/eclipse/microprofile12/metrics/base/BaseMetricsTest.java]: Remove ` gauge` from the end of the 2nd alternative.

**Comment by arjantijms** [comment]
Travis seems to have an off day, but the failure is unrelated to the code.

File: metrics/base-metrics/src/test/java/org/eclipse/microprofile12/metrics/base/BaseMetricsTest.java [hunk 2]
```diff
@@ -55,10 +55,10 @@ public void testBasePrometheusMetrics() throws IOException {
...
+        assertTrue(responseText.contains("# TYPE base:thread_count") || responseText.contains("# TYPE base_thread_count"));
+        assertTrue(responseText.contains("# TYPE base:cpu_system_load_average") || responseText.contains("# TYPE base_cpu_systemLoadAverage"));
+        assertTrue(responseText.contains("# TYPE base:jvm_uptime") || responseText.contains("# TYPE base_jvm_uptime_seconds"));
```
Comment (Ladicek) [inline metrics/base-metrics/src/test/java/org/eclipse/microprofile12/metrics/base/BaseMetricsTest.java]: I'd probably put the 2nd alternative to a 2nd line, like in the previous file.

File: metrics/base-metrics/src/test/java/org/eclipse/microprofile12/metrics/base/BaseMetricsTest.java (hunk 1 above)
Comment (Ladicek) [reply metrics/base-metrics/src/test/java/org/eclipse/microprofile12/metrics/base/BaseMetricsTest.java]: I mean, it obviously _is_ a gauge, so it technically isn't wrong. I'm asking to remove it just for consistency with the other assertions.

File: metrics/base-metrics/src/test/java/org/eclipse/microprofile12/metrics/base/BaseMetricsTest.java [hunk 3]
```diff
@@ -55,10 +55,14 @@ public void testBasePrometheusMetrics() throws IOException {
...
-        assertTrue(responseText.contains("# TYPE base:thread_max_count"));
-        assertTrue(responseText.contains("# TYPE base:thread_count"));
-        assertTrue(responseText.contains("# TYPE base:cpu_system_load_average"));
-        assertTrue(responseText.contains("# TYPE base:jvm_uptime"));
+        assertTrue(responseText.contains("# TYPE base:thread_max_count") 
+                || responseText.contains("# TYPE base_thread_max_count"));
+        assertTrue(responseText.contains("# TYPE base:thread_count") 
+                || responseText.contains("# TYPE base_thread_count"));
+        assertTrue(responseText.contains("# TYPE base:cpu_system_load_average") 
+                || responseText.contains("# TYPE base_cpu_systemLoadAverage"));
```
Comment (Ladicek) [inline metrics/base-metrics/src/test/java/org/eclipse/microprofile12/metrics/base/BaseMetricsTest.java:63]: I was myself wondering if this is correct, so I took a look into the MP Metrics 2.0 spec. It says:

> Metric names with camelCase are no longer converted to snake_case for OpenMetrics output.

So yes, this is correct.

**Review by Ladicek — APPROVED** [review]

**Pull request merged**